pkg/view/router.go

```diff
@@ -1,49 +1,104 @@
 package view
 
 import (
+	"encoding/json"
+	"fmt"
+	"html/template"
 	"net/http"
+	"strconv"
 
 	"github.com/foolin/goview"
 	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+	"github.com/gorilla/csrf"
 	"github.com/scratchdata/scratchdata/pkg/config"
+	"github.com/scratchdata/scratchdata/pkg/destinations"
+	"github.com/scratchdata/scratchdata/pkg/storage"
 	"github.com/scratchdata/scratchdata/pkg/storage/database/models"
-	"github.com/scratchdata/scratchdata/templates"
+	"github.com/scratchdata/scratchdata/pkg/view/templates"
 )
 
+type Connections struct {
+	Destinations []config.Destination
+}
+
+type UpsertConnection struct {
+	Destination config.Destination
+}
+
+type Connect struct {
+	InsertExample string
+	QueryExample  string
+}
+
 type Model struct {
-	Email string
+	CSRFToken        template.HTML
+	Email            string
+	Connect          Connect
+	Connections      Connections
+	UpsertConnection UpsertConnection
 }
 
 func embeddedFH(config goview.Config, tmpl string) (string, error) {
 	bytes, err := templates.Templates.ReadFile(tmpl + config.Extension)
 	return string(bytes), err
 }
 
-func New(c config.DashboardConfig, auth func(h http.Handler) http.Handler) (*chi.Mux, error) {
+func New(
+	storageServices *storage.Services,
+	c config.DashboardConfig,
+	destManager *destinations.DestinationManager,
+	auth func(h http.Handler) http.Handler,
+) (*chi.Mux, error) {
 	r := chi.NewRouter()
 
+	csrfMiddleware := csrf.Protect([]byte(c.CSRFSecret))
```

Comment:
the `/logout` route should just clear this too. Maybe it should just iterate over every cookie and clear it.